### ISSUE TITLE
e2e pilot test logs 399Mbytes upon failure

### ISSUE THREAD

**Issue by ldemailly** [istio/istio#3499]
https://github.com/istio/istio/blame/66aec2e5f7a06eb9f160b5914628bd8ced752640/tests/e2e/tests/pilot/pilot_test.go#L94

$ ls -lh 3459.log.txt 
-rw-r--r--  1 ldemailly  _guest   399M Feb 14 13:30 3459.log.txt

relevant section of the huge log:
```
I0214 17:19:26.980] =================== Deploying infrastructure =====================
I0214 17:19:26.981] (*util.Infra)(0xc4201d4380)({
I0214 17:19:26.981]  Name: (string) (len=12) "(auth infra)",
I0214 17:19:26.981]  KubeConfig: (string) (len=28) "/home/bootstrap/.kube/config",
I0214 17:19:26.981]  KubeClient: (*kubernetes.Clientset)(0xc4200ae690)({
I0214 17:19:26.981]   DiscoveryClient: (*discovery.DiscoveryClient)(0xc4204220a0)({
I0214 17:19:26.981]    restClient: (*rest.RESTClient)(0xc4203ea000)({
I0214 17:19:26.981]     base: (*url.URL)(0xc42025a100)(https://104.154.55.51/),
I0214 17:19:26.981]     versionedAPIPath: (string) (len=1) "/",
I0214 17:19:26.981]     contentConfig: (rest.ContentConfig) {
I0214 17:19:26.981]      AcceptContentTypes: (string) "",
I0214 17:19:26.982]      ContentType: (string) (len=16) "application/json",
I0214 17:19:26.982]      GroupVersion: (*schema.GroupVersion)(0xc420422040)(meta.k8s.io/v1),
I0214 17:19:26.982]      NegotiatedSerializer: (*serializer.negotiatedSerializerWrapper)(0xc4202d3480)({
I0214 17:19:26.982]       info: (runtime.SerializerInfo) {
I0214 17:19:26.982]        MediaType: (string) "",
I0214 17:19:26.982]        EncodesAsText: (bool) false,
I0214 17:19:26.982]        Serializer: (runtime.NoopEncoder) {
I0214 17:19:26.982]         Decoder: (*versioning.codec)(0xc420228880)({
I0214 17:19:26.983]          encoder: (runtime.Encoder) <nil>,
I0214 17:19:26.983]          decoder: (*recognizer.decoder)(0xc420210660)({
I0214 17:19:26.983]           decoders: ([]runtime.Decoder) (len=3 cap=3) {
I0214 17:19:26.983]            (*json.Serializer)(0xc4201eea00)({
I0214 17:19:26.983]             meta: (json.SimpleMetaFactory) {
I0214 17:19:26.983]             },
I0214 17:19:26.983]             creater: (*runtime.Scheme)(0xc4201ee9c0)({
I0214 17:19:26.983]              gvkToType: (map[schema.GroupVersionKind]reflect.Type) (len=322) {
I0214 17:19:26.983]               (schema.GroupVersionKind) authorization.k8s.io/v1, Kind=LocalSubjectAccessReview: (*reflect.rtype)(0x1443220)(v1.LocalSubjectAccessReview),
...
```
including multi megabytes of reflection pointers, freebuf etc etc
```
I0214 17:31:54.364]                     00013be0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
I0214 17:31:54.364]                     00013bf0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
I0214 17:31:54.364]                     00013c00  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
```
which compress well

**Comment by nmittler** [comment]
@ldemailly can we close this?

**Comment by ldemailly** [comment]
@nmittler yes! thanks! next time put "Fixes #3499" in your fixing PR (#3502)
